packages/blade/src/hooks/useControllable.ts

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
```

Comment:
This hook is probably the most important/useful hook :p 

It automatically handles controlled and uncontrolled state while internally giving us the state value so that we can react to the changes. 

For checkbox's use case we want to internally track the `checked` state to be able to render the correct Icon. 

This hook will also be handy in other cases like Input. 
Where you will need to provide a controlled/uncontrolled API to the user but need to keep track of the input's value also internally to show that character count component. 


 
You just pass the props and internally it will return you a piece of state.

```
  const [checkboxState, setCheckboxStateChange] = useControllableState({
    value: isChecked,
    defaultValue: defaultChecked,
    onChange,
  });
  
```

onChange will automatically react to changes, and depending on isChecked & defaultChecked this hook will automatically handle uncontrolled and uncontrolled behaviour.